packages/atomic/src/components/recommendations/atomic-recs-list/atomic-recs-list.tsx

```diff
@@ -187,39 +255,141 @@ export class AtomicRecsList implements InitializableComponent<RecsBindings> {
     }
   }
 
-  private get resultListCommonState(): ResultListCommonState<RecsResult> {
-    return {
-      firstSearchExecuted: this.recommendationListState.searchResponseId !== '',
-      isLoading: this.recommendationListState.isLoading,
-      hasError: this.recommendationListState.error !== null,
-      hasResults: this.recommendationListState.recommendations.length !== 0,
-      results: this.recommendationListState.recommendations,
-      searchResponseId: this.recommendationListState.searchResponseId,
-    };
-  }
-
   private renderHeading() {
     if (!this.label) {
       return;
     }
 
-    const shouldHide =
-      this.resultListCommonState.hasError ||
-      (this.resultListCommonState.firstSearchExecuted &&
-        !this.resultListCommonState.hasResults);
+    if (this.resultListCommonState.hasError) {
+      return;
+    }
+
+    if (
+      this.resultListCommonState.firstSearchExecuted &&
+      !this.resultListCommonState.hasResults
+    ) {
+      return;
+    }
 
     return (
-      <Heading
-        level={this.headingLevel}
-        part="label"
-        class={shouldHide ? 'hidden' : ''}
-      >
+      <Heading level={this.headingLevel} part="label" class="m-0 mb-2">
         {this.bindings.i18n.t(this.label)}
       </Heading>
     );
   }
 
+  private get currentIndex() {
```

Comment:
There is an argument that the pagination functionality/logic of the recommendation component could be moved to the Headless recommendation list controller...

We could also potentially want to do analytics tracking when someone clicks the next/previous button 🤔 Which would be tackled by Headless.

Reply:
Interesting...

Reply:
If that was moved to Headless, then we'd have to deal with two different "number of recommendation" parameters, which I think may get a bit confusing.

I think we should discuss with ML whether this information could be useful to them.

Reply:
I've investigated, having 2 lists saved in headless is a little bizarre... but we could nevertheless log something here even if it's purely visual.